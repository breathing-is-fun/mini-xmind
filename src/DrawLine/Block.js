import React, { Component } from 'react';

import './assets/Block.css';

export default class Block extends Component {
  render() {
    const { top, left, color, className } = this.props;
    const style = { top, left, backgroundColor: color };
    return (
      <div
        className={`block ${className}`}
        style={style}
        onMouseOver={this.props.onMouseOver}
        onMouseOut={this.props.onMouseOut}
      >
        {this.props.children}
      </div>
    );
  }
}
