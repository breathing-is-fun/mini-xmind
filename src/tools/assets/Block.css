.Block {
  position: absolute;
  width: 50px;
  height: 50px;
  line-height: 50px;
  background-color: #000;
  cursor: pointer;
  text-align: center;
  color: #fff;
  z-index: 2;
}