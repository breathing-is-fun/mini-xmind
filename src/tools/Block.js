import React, { Component } from 'react';

import './assets/Block.css';

export default class Block extends Component {
  render() {
    const { top, left, color, className, children, ...rest } = this.props;
    return (
      <div className={`block ${className}`} {...rest}>
        {children}
      </div>
    );
  }
}
