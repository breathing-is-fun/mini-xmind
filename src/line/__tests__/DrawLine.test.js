import React from 'react';
import { render, mount, shallow } from 'enzyme';
import { SteppedLineTo } from '..';
import Block from '../../tools/Block';

describe('DrawLine', () => {
  it('DrawLine render correctly', () => {
    class Demo extends React.Component {
      render() {
        const style = {
          delay: true,
          borderColor: '#ddd',
          borderStyle: 'solid',
          borderWidth: 3,
        };

        return (
          <React.Fragment>
            <Block className="stepped-A" top="50px" left="90px" color="#00f">
              A
            </Block>
            <Block className="stepped-B" top="150px" left="20px" color="#00f">
              B
            </Block>
            <Block className="stepped-C" top="150px" left="90px" color="#00f">
              C
            </Block>
            <Block className="stepped-D" top="150px" left="160px" color="#00f">
              D
            </Block>
            <Block className="stepped-E" top="50px" left="300px" color="#00f">
              E
            </Block>
            <Block className="stepped-F" top="120px" left="300px" color="#00f">
              F
            </Block>
            <SteppedLineTo
              from="stepped-A"
              to="stepped-B"
              fromAnchor="bottom"
              toAnchor="top"
              {...style}
            />
            <SteppedLineTo
              from="stepped-A"
              to="stepped-C"
              fromAnchor="bottom"
              toAnchor="top"
              {...style}
            />
            <SteppedLineTo
              from="stepped-A"
              to="stepped-D"
              fromAnchor="bottom"
              toAnchor="top"
              {...style}
            />
            <SteppedLineTo
              from="stepped-A"
              to="stepped-E"
              fromAnchor="right"
              toAnchor="left"
              orientation="h"
              {...style}
            />
            <SteppedLineTo
              from="stepped-A"
              to="stepped-F"
              fromAnchor="right"
              toAnchor="left"
              orientation="h"
              {...style}
            />
          </React.Fragment>
        );
      }
    }
    const wrapper = mount(<Demo />);

    expect(wrapper).toMatchSnapshot();
  });
});
