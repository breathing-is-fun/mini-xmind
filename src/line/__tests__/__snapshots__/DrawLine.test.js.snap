// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawLine DrawLine render correctly 1`] = `
<Demo>
  <Block
    className="stepped-A"
    color="#00f"
    left="90px"
    top="50px"
  >
    <div
      className="block stepped-A"
    >
      A
    </div>
  </Block>
  <Block
    className="stepped-B"
    color="#00f"
    left="20px"
    top="150px"
  >
    <div
      className="block stepped-B"
    >
      B
    </div>
  </Block>
  <Block
    className="stepped-C"
    color="#00f"
    left="90px"
    top="150px"
  >
    <div
      className="block stepped-C"
    >
      C
    </div>
  </Block>
  <Block
    className="stepped-D"
    color="#00f"
    left="160px"
    top="150px"
  >
    <div
      className="block stepped-D"
    >
      D
    </div>
  </Block>
  <Block
    className="stepped-E"
    color="#00f"
    left="300px"
    top="50px"
  >
    <div
      className="block stepped-E"
    >
      E
    </div>
  </Block>
  <Block
    className="stepped-F"
    color="#00f"
    left="300px"
    top="120px"
  >
    <div
      className="block stepped-F"
    >
      F
    </div>
  </Block>
  <SteppedLineTo
    borderColor="#ddd"
    borderStyle="solid"
    borderWidth={3}
    delay={true}
    from="stepped-A"
    fromAnchor="bottom"
    to="stepped-B"
    toAnchor="top"
  />
  <SteppedLineTo
    borderColor="#ddd"
    borderStyle="solid"
    borderWidth={3}
    delay={true}
    from="stepped-A"
    fromAnchor="bottom"
    to="stepped-C"
    toAnchor="top"
  />
  <SteppedLineTo
    borderColor="#ddd"
    borderStyle="solid"
    borderWidth={3}
    delay={true}
    from="stepped-A"
    fromAnchor="bottom"
    to="stepped-D"
    toAnchor="top"
  />
  <SteppedLineTo
    borderColor="#ddd"
    borderStyle="solid"
    borderWidth={3}
    delay={true}
    from="stepped-A"
    fromAnchor="right"
    orientation="h"
    to="stepped-E"
    toAnchor="left"
  />
  <SteppedLineTo
    borderColor="#ddd"
    borderStyle="solid"
    borderWidth={3}
    delay={true}
    from="stepped-A"
    fromAnchor="right"
    orientation="h"
    to="stepped-F"
    toAnchor="left"
  />
</Demo>
`;
